### PR TITLE
chore(forms): update group forms to new input format

### PR DESCRIPTION
**Pull request by hypeJunction**
Updates group forms to use elgg_view_input()
- [ ] rebase
- [ ] update w/r/t new features in #9106

### CODE REVIEW

File: mod/groups/views/default/groups/edit/access.php [hunk 1]
```diff
@@ -8,128 +8,92 @@
...
+	'id' => 'groups-content-access-mode',
+	'value' => $content_access_mode,
+	'options_values' => [
+		ElggGroup::CONTENT_ACCESS_MODE_UNRESTRICTED => elgg_echo('groups:content_access_mode:unrestricted'),
+		ElggGroup::CONTENT_ACCESS_MODE_MEMBERS_ONLY => elgg_echo('groups:content_access_mode:membersonly'),
+	],
+	'label' => elgg_echo('groups:content_access_mode'),
+	// Warn the user that changing the content access mode to more
+	// restrictive will not affect the existing group content
+	'help' => $entity && $entity->getContentAccessMode() == ElggGroup::CONTENT_ACCESS_MODE_UNRESTRICTED ? elgg_echo('groups:content_access_mode:warning') : '',
```
Comment (mrclay) [inline mod/groups/views/default/groups/edit/access.php]: Ternaries should be short. This should probably be an if/else setting $help.

Comment (mrclay) [inline mod/groups/views/default/groups/edit/access.php]: BTW, what happens if help is set to `""` or `null`? Does it get a wrapper element?

File: mod/groups/views/default/groups/edit/access.php [hunk 2]
```diff
@@ -8,128 +8,92 @@
...
-<?php
+	'disabled' => $entity && $entity->access_id === $entity->group_acl,
+]);
 
 if ($entity && ($owner_guid == elgg_get_logged_in_user_guid() || elgg_is_admin_logged_in())) {
-	$members = array();
+	$members = [];
 
 	$dbprefix = elgg_get_config('dbprefix');
-	$options = array(
```
Comment (mrclay) [inline mod/groups/views/default/groups/edit/access.php]: In the future please resist the urge to change stuff like this. It doesn't add readability, it adds noise to the PR and git history.

File: mod/groups/views/default/groups/edit/profile.php [hunk 3]
```diff
@@ -8,40 +8,37 @@
...
 // show the configured group profile fields
 foreach ((array)$group_profile_fields as $shortname => $valtype) {
-	if ($valtype == "hidden") {
-		echo elgg_view("input/{$valtype}", array(
-			"name" => $shortname,
-			"value" => elgg_extract($shortname, $vars),
-		));
+	$value = elgg_extract($shortname, $vars);
+
+	if ($valtype == 'hidden') {
```
Comment (mrclay) [inline mod/groups/views/default/groups/edit/profile.php]: Should we just teach elgg_view_input() not to wrap "hidden"?

File: mod/groups/views/default/groups/edit/tools.php [hunk 4]
```diff
@@ -8,20 +8,27 @@
...
 			'default' => 'no',
 			'checked' => ($value === 'yes') ? true : false,
 			'label' => $group_option->label
-		)));
+		]));
 	}
+}
+
+if ($input) {
+	echo elgg_view('elements/forms/field', [
```
Comment (mrclay) [inline mod/groups/views/default/groups/edit/tools.php]: Would it be a pain to convert this view to input/checkboxes? Would that work?

File: mod/groups/views/default/groups/edit/access.php (hunk 2 above)
Comment (hypeJunction) [inline mod/groups/views/default/groups/edit/access.php]: Sorry, new bad habit, of using IDE to auto-change everything to short syntax...

File: mod/groups/views/default/groups/edit/profile.php (hunk 3 above)
Comment (hypeJunction) [inline mod/groups/views/default/groups/edit/profile.php]: yes

File: mod/groups/views/default/groups/edit/tools.php (hunk 4 above)
Comment (hypeJunction) [inline mod/groups/views/default/groups/edit/tools.php]: we would need to teach input/checkboxes to render custom names

Comment (mrclay) [inline mod/groups/views/default/groups/edit/tools.php]: Eh, nevermind. :)

Comment (hypeJunction) [inline mod/groups/views/default/groups/edit/tools.php]: We can create `input/groups/tools` view that would take a list of tools in and spit out checkboxes

File: mod/groups/views/default/groups/edit/access.php (hunk 1 above)
Comment (hypeJunction) [inline mod/groups/views/default/groups/edit/access.php]: `if (!$help) return;` so anything false is not wrapped

File: mod/groups/views/default/groups/edit/profile.php (hunk 3 above)
Comment (hypeJunction) [inline mod/groups/views/default/groups/edit/profile.php]: PR #9106

File: mod/groups/views/default/groups/edit/tools.php (hunk 4 above)
Comment (mrclay) [inline mod/groups/views/default/groups/edit/tools.php]: SGTM.

**Comment by hypeJunction** [comment]
If anyone gets back to this before I do, I have isolated all the group-specific fields into their own views: https://github.com/hypeJunction/Elgg-prototyper_group/tree/master/views/default/input/groups

**Comment by jdalsem** [comment]
@hypeJunction do you want something more with this PR? i think it can go in as is... maybe another PR for the input/groups/tools...

**Comment by mrclay** [comment]
I added a couple TODOs.

**Comment by jdalsem** [comment]
@hypeJunction why was this closed?

**Comment by hypeJunction** [comment]
I was just clearing up all the stuff that I am no longer planning to work on.

**Pull request closed without merging**